File: 백준/Silver/16931. 겉넓이 구하기/겉넓이 구하기.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "ex.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);

let side1 = 0;
let side2 = 0;
let side3 = n * m;

const arr = [];
for (let i = 1; i <= n; i++) {
  arr.push(input[i].split(" ").map(Number));
}

for (let i = 0; i < n; i++) {
  for (let j = 0; j < m; j++) {
    if (j === 0) {
      side1 += arr[i][j];
    } else {
      if (arr[i][j] > arr[i][j - 1]) {
        side1 += arr[i][j] - arr[i][j - 1];
      }
    }
  }
}

for (let i = 0; i < n; i++) {
  for (let j = 0; j < m; j++) {
    if (i === 0) {
      side2 += arr[i][j];
    } else {
      if (arr[i][j] > arr[i - 1][j]) {
        side2 += arr[i][j] - arr[i - 1][j];
      }
    }
  }
}

console.log((side1 + side2 + side3) * 2);

File: 백준/Silver/16931. 겉넓이 구하기/겉넓이 구하기.py
N, M = map(int, input().split())

arr = []
left = 0
bottom = 0

for i in range(N):
    data = list(map(int, input().split()))
    arr.append(data)

# 왼쪽
for i in range(N):
    for j in range(M):
        if j == 0:
            left += arr[i][j]
        else:
            if arr[i][j-1] < arr[i][j]:
                left += arr[i][j] - arr[i][j-1]

# 아래쪽
for j in range(M):
    for i in range(N):
        if i == 0:
            bottom += arr[i][j]
        else:
            if arr[i-1][j] < arr[i][j]:
                bottom += arr[i][j] - arr[i-1][j]

print((left+bottom+N*M)*2)
